cli/cmd/inject.go

```diff
@@ -171,20 +171,25 @@ func (rt resourceTransformerInject) transform(bytes []byte) ([]byte, []inject.Re
 	reports := []inject.Report{*report}
 
 	if conf.IsService() {
-		opaquePortsAnnotations := map[string]string{}
-		if opaquePorts, ok := rt.overrideAnnotations[k8s.ProxyOpaquePortsAnnotation]; ok {
-			opaquePortsAnnotations[k8s.ProxyOpaquePortsAnnotation] = opaquePorts
-			b, err := conf.AnnotateService(opaquePortsAnnotations)
-			return b, reports, err
+		opaquePorts, ok := rt.overrideAnnotations[k8s.ProxyOpaquePortsAnnotation]
+		if ok {
+			annotations := map[string]string{k8s.ProxyOpaquePortsAnnotation: opaquePorts}
+			bytes, err = conf.AnnotateService(annotations)
+			report.Annotated = true
 		}
-		return bytes, reports, nil
+		return bytes, reports, err
 	}
-
 	if rt.allowNsInject && conf.IsNamespace() {
```

Comment:
Nope this is unrelated to opaque ports. This already exists and is for adding the `linkerd.io/inject: enabled` annotation to namespaces. Check out `func AnnotateNamespace`.

When this is called through the CLI, `allowNsInject` is set to true ([1](https://github.com/linkerd/linkerd2/blob/0ff5528085a998a643f87b1c2973987849afe0dc/cli/cmd/inject.go#L106-L113)) so that's how this block would be entered.